staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -0,0 +1,1019 @@
+	ca.queue.Forget(attRef)
+}
+
+func (ca *ConnectionAgent) onLocalAttAdded(obj interface{}) {
+	localAtt := obj.(*netv1a1.NetworkAttachment)
+	glog.V(5).Infof("Notified of NetworkAttachment %#+v addition to local NetworkAttachments cache\n", localAtt)
+	localAttRef := ca.fromAttToEnquableRef(localAtt)
+	ca.queue.Add(localAttRef)
+}
+
+// TODO make the enqueuing conditional on a check on whether the update is relevant (e.g. an update which adds a metadata
+// label is completely irrelevant and should not trigger an enqueuing)
+func (ca *ConnectionAgent) onLocalAttUpdated(oldObj, newObj interface{}) {
+	oldLocalAtt := oldObj.(*netv1a1.NetworkAttachment)
+	newLocalAtt := newObj.(*netv1a1.NetworkAttachment)
+	glog.V(5).Infof("Notified of NetworkAttachment update from %#+v to %#+v in local NetworkAttachments cache\n", oldLocalAtt, newLocalAtt)
+
+	newLocalAttRef := ca.fromAttToEnquableRef(newLocalAtt)
+	ca.queue.Add(newLocalAttRef)
+	if newLocalAtt.Spec.VNI != oldLocalAtt.Spec.VNI {
+		// A reference is made up by a namespaced name, a vni, and a locality flag. Changes to namespace and name are not possible:
+		// that's equivalent to creating a new, independent network attachment. Changes to the node where the network attachment runs
+		// won't result in an update but in a deletion because the informer this handler is registered to has a field selector on the
+		// value of this node. The only changes to a NetworkAttachment that cause an update and a change in the fields of its enquable
+		// reference are those to its VNI value. In this case, the old object and the updated object have different references. We enqueue
+		// both so that not only the new data structures (e.g. Network Interface) are allocated, but the old ones are deallocated (by two
+		// separate worker threads).
+		oldLocalAttRef := ca.fromAttToEnquableRef(oldLocalAtt)
+		ca.queue.Add(oldLocalAttRef)
+	}
+}
+
+func (ca *ConnectionAgent) onLocalAttRemoved(obj interface{}) {
+	peeledObj := peel(obj)
+	localAtt := peeledObj.(*netv1a1.NetworkAttachment)
+	glog.V(5).Infof("Notified of NetworkAttachment %#+v removal from local NetworkAttachments cache\n", localAtt)
+	localAttRef := ca.fromAttToEnquableRef(localAtt)
+	ca.queue.Add(localAttRef)
+}
+
+func (ca *ConnectionAgent) processLocalAtt(attRef attQueueRef) error {
+	localAtt, err := ca.localAttsLister.NetworkAttachments(attRef.nsn.Namespace).Get(attRef.nsn.Name)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		// This should never happen.  No point in retrying.
+		glog.Errorf("NetworkAttachments Lister failed to lookup %s/%s: %s\n",
+			attRef.nsn.Namespace, attRef.nsn.Name, err.Error())
+		return nil
+	}
+	if err != nil && k8serrors.IsNotFound(err) {
+		// If we're here the attachment has been deleted
+		err = ca.releaseLocalAttResources(attRef)
+		return err
+	}
+	if err == nil && attRef.vni != localAtt.Spec.VNI {
```

Comment:
I see two IF/THEN statements here with the same THEN.  I suggest combining them.

Reply:
Reduced two IF/THEN to one in 06d37b6.